### DISCUSSION TITLE
Merging clns files

### DISCUSSION THREAD

**Discussion by ig005bvr** [milaboratory/mixcr#1584] [Q&A] [answered]
Hello! Apologies if this is a basic question, but I've been unable to locate an answer in the documentation. I'm trying to combine two .clns files that were generated by running the analyze command. These files are from two separate single-cell sequencing experiments of the same sample. I'm unable to merge the original fastq files due to the potential for barcode overlap. How can I achieve this?

**Comment by mizraelson** [comment]
Hi,
What commands did you for analyzing the data?

**Reply by ig005bvr** [reply]
For example,
I need to merge two .clns data from mixcr analyze 10x-sc-xcr-vdj

**Reply by mizraelson** [reply, marked as answer]
I see, so the correct approach would be as follows. Suppose you have several pairs of FASTQ files from different experiments or replicates for the same biological sample, and the barcodes might overlap:

```
sample1_experiment1_R1.fastq.gz
sample1_experiment1_R2.fastq.gz
sample1_experiment2_R1.fastq.gz
sample1_experiment2_R2.fastq.gz
```

Then, the command would be:

```
mixcr analyze 10x-sc-xcr-vdj \
--species hsa \
--pattern "^(CELL2:N{16})(UMI:N{10})\^(R2:*)" \
sample1_{{CELL1:a}}_{{R}}.fastq.gz
output
```

This way, MiXCR will use part of the input file name as a barcode too, resolving the case of two identical barcodes in different samples.

**Reply by ig005bvr** [reply]
Thank you!
